tests/jax/integration/trainer_test.py

```diff
@@ -0,0 +1,201 @@
+import functools
```

Comment:
please add the license at the top of this file

Reply:
Done!